lib/waterline/query/dql/destroy.js

```diff
@@ -74,6 +74,8 @@ module.exports = function(criteria, cb) {
         break;
```

Comment:
Any ideas why the adapter returns `undefined` PKs in `result`?

We probably need this change to sanitise adapters output but it looks like the root cause of this issue may be caused by the adapters themselves.